modules/apps/data-engine/data-engine-rest-impl/src/main/java/com/liferay/data/engine/rest/internal/dto/v2_0/util/DataDefinitionFieldUtil.java

```diff
@@ -308,15 +308,20 @@ else if (Objects.equals(
 			}
 		}
 
-		if (Validator.isNotNull(customProperties.get("ddmStructureId")) &&
-			Validator.isNull(customProperties.get("ddmStructureKey"))) {
+		String ddmStructureId = GetterUtil.getString(
+			customProperties.get("ddmStructureId"));
 
+		if (Validator.isNotNull(ddmStructureId)) {
 			DDMStructure ddmStructure =
 				ddmStructureLocalService.fetchDDMStructure(
-					GetterUtil.getLong(customProperties.get("ddmStructureId")));
+					GetterUtil.getLong(ddmStructureId));
 
 			if (ddmStructure != null) {
-				customProperties.put(
+				customProperties.putIfAbsent(
+					"externalReferenceCode",
+					ddmStructure.getExternalReferenceCode());
+
+				customProperties.putIfAbsent(
 					"ddmStructureKey", ddmStructure.getStructureKey());
```

Comment:
The order does not matter in these calls, right? If it is correct, can you sort and remove the breaking line?
```
customProperties.putIfAbsent(
	"ddmStructureKey",
	ddmStructure.getStructureKey());
customProperties.putIfAbsent(
	"externalReferenceCode", ddmStructure.getExternalReferenceCode());
```